### DISCUSSION TITLE
How do I get all value from multiple user attributes?

### DISCUSSION THREAD

**Discussion by ehsanr1366** [keycloak/keycloak#29168] [Q&A]
I have an attribute like this:

`services= test1##test2##test3`

![Uploading image.png…]()

and I want to get all values of services in freemarker. So I use this code:

```
<#if account.attributes["services"]?has_content>				    
    - ${account.attributes["services"]}
</#if>
```

This code only print first item: `test1`

Also I use this: account.attributes.services, but I can not get all values.

So how di I get all items?

**Comment by codespearhead** [comment]
I'm not sure I understand the question: the following Freemarker snippet works as expected.

## Input (try it [here](https://try.freemarker.apache.org/))
```freemarker
<!DOCTYPE html>
<html>
<head>
    <title>Keycloak Attribute Display</title>
</head>
<body>

<#assign account = {
  "attributes": {
    "services": "test1##test2##test3"
  }
}>

<h1>Original FreeMarker code</h1>

<#if account.attributes["services"]?has_content>				    
  ${account.attributes["services"]}
</#if>

<h1>Corrected FreeMarker code</h1>

<#if account.attributes["services"]?has_content>
  <#assign servicesList = account.attributes["services"]?split("##")>
    <ul>
      <#list servicesList as service>
        <li>${service}</li>
      </#list>
    </ul>
</#if>

</body>
</html>
```

## Output

```html
<!DOCTYPE html>
<html>
<head>
    <title>Keycloak Attribute Display</title>
</head>
<body>

<h1>Original FreeMarker code</h1>

  test1##test2##test3

<h1>Corrected FreeMarker code</h1>

    <ul>
        <li>test1</li>
        <li>test2</li>
        <li>test3</li>
    </ul>

</body>
</html>
```

**Comment by ehsanr1366** [comment]
Thank you for your reply.

But if you pay attention to my code, I had stated in my question that the output of the following command:

```${account.attributes["services"]}```

is only : `test1` not `test1##test2##test3`

Also in logs, I got this:
```
There are more values for attribute 'services' of user 'test' . Will display just first value
```

If I use any other option instead of `##`, such as a comma, the split command works fine. Apparently, this problem occurs because keycloak uses `##` to separate lists, so there must be another solution.

On the other hand, if in java files, it is easy to get a list of output from this attribute.

```context.getUser().getAttributes().get("services");```
